src/guiapi/include/Rect16.h

```diff
@@ -0,0 +1,279 @@
+    uint16_t Width() const { return width_; };
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Object accessor to read the height of current rectangle
+    ///
+    /// @return Height of the rectangle.
+    uint16_t Height() const { return height_; };
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Object accessor to read the size of current rectangle
+    ///
+    /// @return Size of the rectangle.
+    size_ui16_t Size() const { return size_ui16_t { width_, height_ }; }
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Object accessor to read the top-left corner of current rectangle
+    ///
+    /// @return Top-left corner of the rectangle.
+    point_i16_t TopLeft() const { return top_left_; };
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Object accessor to read the bottom-right of current rectangle
+    ///
+    /// @return Bottom-right of the rectangle.
+    point_i16_t BottomRight() const {
+        return {
+            static_cast<int16_t>(top_left_.x + width_),
+            static_cast<int16_t>(top_left_.y + height_)
+        };
+    };
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Determines if the given point is placed inside of the curent rectangle
+    ///
+    /// @param[in] point Point given to check
+    /// @return Return true if the point is in, false otherwise.
+    bool Contain(point_i16_t point) const {
+        return point.x >= top_left_.x
+            && point.x <= (top_left_.x + width_)
+            && point.y >= top_left_.y
+            && point.y <= (top_left_.y + height_);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Determines the rectangle structure that represents the intersection
+    /// of two rectangles.
+    ///
+    /// @param[in] rect Given rectangle to check for intersection
+    /// @return Return a rectangle that represents the intersection of current
+    /// and given.
+    Rect16 Intersection(Rect16 const &) const;
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Determines the rectangle structure that represents the union of
+    /// two rectangles.
+    ///
+    /// @param[in] rect Given rectangle to united
+    /// @return Return a rectangle that represents the union of current
+    /// and given.
+    Rect16 Union(Rect16 const &) const;
+
+    ////////////////////////////////////////////////////////////////////////////
+    /// @brief Determines if the given rectangle has intersection with the current
+    /// one
+    /// @param[in] rect Rectangle given to check
+    /// @return Return true if the rectangles has intersection, false otherwise.
+    bool HasIntersection(Rect16 const &) const;
```

Comment:
missing method if entire rectangle fits inside this one
`bool IsSubrectangle(Rect16 const &) const;` or bool `Contain(Rect16 const &) const;`